src/store/getters.js

```diff
@@ -31,6 +31,9 @@ export const getters = {
 	getAccount: (state) => (id) => {
 		return state.accounts[id]
 	},
+	getAccountSettings: (state) => (id = null) => {
+		return id ? state.accountSettings[id] : state.accountSettings
```

Comment:
I'm not sure if we also need the variant with id=null. Can we make the id mandatory and always return `state.accountSettings[id]`

Reply:
Hm, I am using it [here](https://github.com/nextcloud/mail/pull/3952/files#diff-e01eb159ce5ee0617f7ca141ba6abc5eaf0dd0e4e9c89bfd77979ff31f63c0b6R153) to grab the settings for all accounts to store them. Maybe a separate method like `getAllAccountSettings` for clarity's sake?

Reply:
Oh I see. I didn't consider that `savePreference` is *global*, as in that it's not specific for one account. Hence you have to send this in a batch.

But yes, let's have a getter for this specific object and one for each account's settings for clarity :+1:

Reply:
Hm, so... I wasn't actually using `getAccountSettings()` for anything other than fetching ALL account settings after a refactor yesterday. I renamed the method to `getAllAccountSettings()` in order to be specific and removed the parameter. If a need for something like `getAccountSettings(account.id)` comes up, we could add it down the line. Right now it would be unused.

Reply:
Makese sense, thanks